JKNIG-1 Create project structure for backend Java application

ai-junior-developer opened the pull request:
## Summary
- Maven-based Java 21 project using Spring Boot 3.4.4 created for a books store backend.
- Project structure includes Book domain model, a BookService returning dummy data, and a BookController REST endpoint `/api/books` for fetching books.
- Added `.gitignore`, `README.md`, and `CHANGELOG.md`.

### Technical Details
- **Group ID:** com.eugenscobich
- **Artifact ID:** ai-demo-project
- **Base package:** com.eugenscobich.ai.demo.project
- **Main endpoint:** `GET /api/books` returns a list of dummy books.
- Build and run: `mvn clean package` or `mvn spring-boot:run`

This sets up the initial backend skeleton for further development.

ThreadId:[91b9ba1e-ac97-44a3-87fd-169974269828]

Commit:
JKNIG-1 Create project structure for backend Java application with Spring Boot.
- Added Maven project (Java 21, Spring Boot 3.4.4)
- Implemented Book model, BookService with dummy data, and BookController with /api/books endpoint
- Added README.md, CHANGELOG.md, and .gitignore

eugenscobich commented on src/main/java/com/eugenscobich/ai/demo/project/controller/BookController.java:
```diff
@@ -0,0 +1,23 @@
+public class BookController {
+    private final BookService bookService;
+
+    public BookController(BookService bookService) {
```
Use lombok to generate this constructor

ai-junior-developer replied:
Task was assigned to AI Junior Developer.
Links:
[AI Junior Developer Console](https://jknights.eugenscobich.com/91b9ba1e-ac97-44a3-87fd-169974269828/messages)
[OpenAI Platform](https://platform.openai.com/playground/assistants?assistant=null&thread=91b9ba1e-ac97-44a3-87fd-169974269828)

ai-junior-developer replied:
Message: 
- Added Lombok as a dependency in pom.xml.
- Refactored BookController to use Lombok's @RequiredArgsConstructor for constructor generation.
- Updated CHANGELOG.md to reflect the change.
- Committed and pushed the changes to the remote repository.

The code now uses Lombok to generate the constructor in BookController as requested.

Commit:
JKNIG-1 Use Lombok to generate BookController constructor
- Added Lombok dependency
- Refactored BookController to use @RequiredArgsConstructor
- Updated CHANGELOG.md
